create_heatmaps/base.py

```diff
@@ -119,41 +145,109 @@ def create_heatmaps(self, output_paths, mjd_minmaxes, fit_on_full_lc=True):
-                    total += v
-                f.write("done: {}\n".format(total))
-                f.write("removed: {}\n".format(str(self.removed_by_type).replace("'", "")))
-
+                    
+            # - - - -
+            logging.info(f"job {self.index}: Finsished processing " \
+                         f"{n_lc_write} of {n_lc_read} light curves.")
+
+
+            # xxxxxxxxx mark delete Mar 3 2024 xxxxxxxxx
+            # RK - collisions reading/writing to same finished_filenames file is likely
+            #   leading to rate (1-2%) of tasks crashing. This information is
+            #    contained in the new heatmap*summary file per task, where there is
+            #    no write collision.
+            do_legacy_finished_filenames = False
+            if do_legacy_finished_filenames:
+                if not os.path.exists(self.finished_filenames_path):
+                    pd.DataFrame({"filenames": [os.path.basename(self.metadata_path)]}).to_csv(self.finished_filenames_path, index=False)
+                else:
+                    finished_filenames = pd.read_csv(self.finished_filenames_path)
+                    finished_filenames = pd.concat([finished_filenames, 
+                                                    pd.DataFrame({"filenames": [os.path.basename(self.metadata_path)]})] )
+                    finished_filenames.to_csv(self.finished_filenames_path, index=False)
+            # xxxxxxxx end mark delete xxxxxxxxxxxx
+
+
+            # - - - - - 
+            # write REPORT information to done.log file :
+            self.write_done_file_legacy(output_path)  # original/unformatted
+            self.write_summary_file(heatmap_file)     # Mar 2024 RK - formatted summary 
+            return
+            
+        
+    # ================================================
     # HELPER FUNCTIONS
+    # ================================================
+
+    def write_summary_file(self, heatmap_file):
+        
+        # Created Mar 1 2024 by R.Kessler
+
+        heatmap_file_base = os.path.basename(heatmap_file)
+
+        # summary file name is heatmap file name with .tfrecord replaced by .summary
+        summary_file = heatmap_file.split('.')[0] + '.summary'
+        
+        # get process time for this bundle of heatmaps.
+        t_proc_minutes  = (time.time() - self.t_start )/60.0
+
+        with open(summary_file,"wt") as s:
+            s.write(f"PROGRAM_CLASS:  CreateHeatmaps\n")
+            s.write(f"SURVEY:         {self.survey} \n")
+            s.write(f"HEATMAP_FILE:   {heatmap_file_base}\n")
+            s.write(f"JOBID:          {self.index}\n")
+            s.write(f"CPU:            {t_proc_minutes:.2f}            # minutes \n")
+            s.write(f"LCDATA_PATH:    {self.lcdata_path} \n")
+
+            s.write(f"NLC: \n")
+            for lctype, n_lc in self.done_by_type.items():
+                s.write(f"  {lctype}:  {n_lc:6d}       # TYPE: NLC\n")
+
+            # cpu ??
```

Comment:
should these comments be deleted?

Reply:
If referring to "# TYPE: NLC", this is a comment for humans looking at the output summary file ... it's not a code comment,